rv-monitor/src/test/java/com/runtimeverification/rvmonitor/examples/java/JavaSpecificSyntaxTests.java

```diff
@@ -0,0 +1,88 @@
+import com.runtimeverification.rvmonitor.util.RVMException;
+import org.junit.Test;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+
+public class JavaSpecificSyntaxTests {
+
+    private static final String errIncorrectArgNum = "Unexpected number of parameters";
+    private static final String errIncorrectGenericType = "Unexpected number of parameters";
+
+    private File fromResources(Path filePath) {
+        ClassLoader classLoader = getClass().getClassLoader();
+
+        return new File(classLoader.getResource(filePath.toString()).getFile());
+    }
+
+    private EventDefinitionExt getEvent(String folder, String name) {
+        RVMSpecFileExt parsedFile = null;
+        try {
+            parsedFile = JavaParserAdapter.parse(fromResources(Paths.get(folder, name)));
+            assertNotNull("Parser neither parsed file nor threw exception", parsedFile);
+
+        } catch (RVMException e) {
+            fail("Unexpected ParseException" + e.getMessage());
+        }
+
+        List<RVMonitorSpecExt> specs = parsedFile.getSpecs();
+        assertEquals("Unexpected number of Spec ASTs", 1, specs.size());
+        RVMonitorSpecExt spec = specs.get(0);
+
+        List<EventDefinitionExt> events = spec.getEvents();
+        assertEquals("Unexpected number of Event ASTs", 1, events.size());
+
+        return events.get(0);
+    }
+
+    @Test
+    public void testGenericsInAdviceParams() {
+        EventDefinitionExt testEvent = getEvent("java", "AdviceParamsWithGenerics.rvm");
+        RVMParameters adviceParams = testEvent.getParameters();
+
+        assertEquals(errIncorrectArgNum, 1, adviceParams.size());
+        assertEquals(errIncorrectGenericType
+                , "List<Bar>"
+                , adviceParams.get(0).getType().toString());
+    }
+
+    @Test
+    public void testGenericsInAdviceParamsMultiple() {
+        EventDefinitionExt testEvent = getEvent("java", "AdviceParamsWithGenericsMultiple.rvm");
+        RVMParameters adviceParams = testEvent.getParameters();
+
+        assertEquals(errIncorrectArgNum, 2, adviceParams.size());
+        assertEquals(errIncorrectGenericType
+                , "List<Bar>"
+                , adviceParams.get(0).getType().toString());
+
+        assertEquals(errIncorrectGenericType
+                , "List<Foo>"
+                , adviceParams.get(1).getType().toString());
+    }
+
+
+    @Test
+    public void testGenericsInAdviceParamsNested() {
+        EventDefinitionExt testEvent = getEvent("java", "AdviceParamsWithGenericsNested.rvm");
+        RVMParameters adviceParams = testEvent.getParameters();
+
+        assertEquals(errIncorrectArgNum, 1, adviceParams.size());
+        assertEquals(errIncorrectGenericType
+                , "List<List<Bar>>"
+                , adviceParams.get(0).getType().toString());
+
+    }
```

Comment:
Some of the types that are being parsed in this PR are not tested, e.g., WildCardType, VoidType. Please add tests for those.

Also, can you please add a few tests for non-generic parameters, so we know those work as well?